Updates to KES docs for config reference (#48)
minio/kes-docs: content/integrations/hashicorp-vault-keystore.md

   ```
   {{< /admonition >}}

   **Containers**

   The following instructions use [Podman](https://podman.io/) to manage the containers.
   You can also use Docker.

   Modify addresses and file paths as needed for your deployment.

   ```sh {.copy}
   sudo podman pod create  \
     -p 9000:9000 -p 9001:9001 -p 7373:7373  \
     -v ~/minio-kes-vault/certs:/certs  \
     -v ~/minio-kes-vault/minio:/mnt/minio  \
     -v ~/minio-kes-vault/config:/etc/default/  \
     -n minio-kes-vault
   
   sudo podman run -dt  \
     --cap-add IPC_LOCK  \
     --name kes-server  \
     --pod "minio-kes-vault"  \
     -e KES_SERVER=https://127.0.0.1:7373  \
     -e KES_CLIENT_KEY=/certs/kes-server.key  \
     -e KES_CLIENT_CERT=/certs/kes-server.cert  \
     quay.io/minio/kes:2024-01-11T13-09-29Z server  \
       --auth  \
       --config=/etc/default/kes-config.yaml  \
   
   sudo podman run -dt  \
     --name minio-server  \
     --pod "minio-kes-vault"  \
     -e "MINIO_CONFIG_ENV_FILE=/etc/default/minio"  \
     quay.io/minio/minio:RELEASE.2024-01-31T20-20-33Z server  \
       --console-address ":9001"
   ```

   You can verify the status of the containers using the following command.

>    The command should show three pods, one for hte Pod, one for KES, and one for MinIO.

Reviewer: ```suggestion
   The command should show three pods, one for the Pod, one for KES, and one for MinIO.
```

Author: 👍

| `log`                         | Enable or disable output for `error` and `audit` type logging events to the console. |
| `keystore.vault.endpoint` | The Vault endpoint. For example, `https://127.0.0.1:8200` |
| `keystore.vault.engine` | The path of the K/V engine. For example, `secrets`. <br><br> If empty, defaults to: `kv` (Vault default).  |
| `keystore.vault.version` | The K/V engine version, either `v1` or `v2`. `v1` engine is recommended. |
| `keystore.vault.namespace` | An optional [Vault namespace](https://www.vaultproject.io/docs/enterprise/namespaces/index.html). |
| `keystore.vault.prefix` | An optional K/V prefix. The server stores keys under this prefix. |
| `keystore.vault.transit` | Optionally encrypt keys stored on the K/V engine with a Vault-managed key. |
| `keystore.vault.transit.engine` | The path of the transit engine. For example, `my-transit`. <br><br> If empty, it defaults to: `transit` (Vault default). |
| `keystore.vault.transit.key` | The key name to use to encrypt entries stored on the K/V engine. |
| `keystore.vault.approle` | [AppRole credentials](https://www.vaultproject.io/docs/auth/approle.html). |

> | `keystore.vault.approle.namespace` | Optional Vault namespace used just for authentication. A single `/` is an alias for the Vault `root` namespace. |

Reviewer: another occurrence needing the `just`->`only` and `/` treatment from elsewhere

when I'm done reading I'll go find that and suggest it here

Reviewer: ```suggestion
| `keystore.vault.approle.namespace` | Optional Vault namespace used only for authentication. For the Vault root namespace, use "/". |
```

| `keystore.vault.approle.engine` | The path of the AppRole engine. For example, `authenticate`. <br><br> If empty, defaults to: `approle` (Vault default). |
| `keystore.vault.approle.id` | Your AppRole Role ID |
| `keystore.vault.approle.secret` | Your AppRole ID's secret |
| `keystore.vault.kubernetes` | [Kubernetes credentials](https://www.vaultproject.io/docs/auth/kubernetes). |

> | `keystore.vault.kubernetes.namespace` | Optional Vault namespace used just for authentication. A single `/` is an alias for the Vault `root` namespace. |

Reviewer: same

Reviewer: ```suggestion
| `keystore.vault.kubernetes.namespace` | Optional Vault namespace used only for authentication. For the Vault root namespace, use "/". |
```
